i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -229,4 +425,573 @@ mod tests {
+            ("src/SUMMARY.md:3", "The Baz Subsection"),
+            ("src/foo.md:1", "How to Foo"),
+            ("src/foo.md:3", "The first paragraph about Foo."),
+            ("src/foo/bar.md:1", "How to Bar"),
+            ("src/foo/bar.md:3", "The first paragraph about Bar."),
+            ("src/foo/bar/baz.md:1", "How to Baz"),
+            ("src/foo/bar/baz.md:3", "The first paragraph about Baz."),
+        ];
+
+        let message_tuples = catalog
+            .messages()
+            .map(|msg| (msg.source(), msg.msgid()))
+            .collect::<Vec<(&str, &str)>>();
+
+        for message_tuple in expected_message_tuples {
+            assert!(message_tuples.contains(&message_tuple));
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_split_catalog_nested_directories_depth_1() -> anyhow::Result<()> {
+        let (ctx, _tmp) = create_render_context(&[
+            ("book.toml", "[book]"),
+            (
+                "src/SUMMARY.md",
+                "# Summary\n\n\
+                 - [Intro](index.md)\n\
+                 # Foo\n\n\
+                 - [The Foo Chapter](foo.md)\n\
+                 \t- [The Bar Section](foo/bar.md)\n\
+                 \t\t- [The Baz Subsection](foo/bar/baz.md)\n\
+                 - [Foo Exercises](exercises/foo.md)",
+            ),
+            (
+                "src/index.md",
+                "# Intro to X\n\
+                 \n\
+                 In this course, you will learn X.\n",
+            ),
+            (
+                "src/foo.md",
+                "# How to Foo\n\
+                 \n\
+                 The first paragraph about Foo.\n",
+            ),
+            (
+                "src/foo/bar.md",
+                "# How to Bar\n\
+                 \n\
+                 The first paragraph about Bar.\n",
+            ),
+            (
+                "src/foo/bar/baz.md",
+                "# How to Baz\n\
+                 \n\
+                 The first paragraph about Baz.\n",
+            ),
+            (
+                "src/exercises/foo.md",
+                "# Exercises on Foo\n\
+                 \n\
+                 Try out this Foo example.\n",
+            ),
+        ])?;
+
+        let catalogs = create_catalogs(&ctx, 1)?;
+
+        let message_tuples = catalogs
+            .into_iter()
+            .map(|(file_rel_path, catalog)| (file_rel_path, msgids_from_catalog(catalog)))
+            .collect::<Vec<(PathBuf, Vec<String>)>>();
+
+        let expected_message_tuples = HashMap::from([
+            (
+                PathBuf::from("summary"),
+                vec![
+                    "Intro",
+                    "Foo",
+                    "The Foo Chapter",
+                    "The Bar Section",
+                    "The Baz Subsection",
+                    "Foo Exercises",
+                    "Intro to X",
+                    "In this course, you will learn X.",
+                ],
```

Comment:
Oh nice, this is much closer to what I suggested above :smile: 

I think it could be made a tiny bit clearer if you tested against the sources.

Reply:
I re-worked this to test against the source string instead like you suggested above